### PR TITLE
Landing Page Adopters Carousel [Draft]

### PR DESCRIPTION
**Pull request by juozasg**
Static draft preview of the adopters layout on the landing page. Carousel controls (Next/Previous buttons and Current Item Indicator) commented out to make it very simple and clean. Clicking the item links to the /adopters Page. 

What remains to be done:
* Clean up and factor out the markup
* Integrate with a YAML data source
* Modify/clone existing python scripts to generate YAML data weekly.  
* Update this to work with Landing Page section header color fixes that need to happen first

One idea: instead of generating 5 items weekly we could include all the items with logos and shuffle them randomly. 

@dholbach @hiddeco

### CODE REVIEW

File: content/en/_index.html [hunk 1]
```diff
@@ -97,6 +97,54 @@ <h2 class="section-label">GitOps at your fingertips</h2>
...
+        <li data-target="#carousel-selected-indicator" data-slide-to="1"></li>
+        <li data-target="#carousel-selected-indicator" data-slide-to="2"></li>
+        <li data-target="#carousel-selected-indicator" data-slide-to="3"></li>
+        <li data-target="#carousel-selected-indicator" data-slide-to="4"></li>
+      </ol>
+      -->
+      <div class="carousel-inner">
+        <a href="/adopters">
+          <div class="carousel-item active">
+            <img class="d-block w-100" src="https://fluxcd.io/img/logos/sageailabs.png" alt="First slide">
```
Comment (dholbach) [inline content/en/_index.html:117]:
```suggestion
            <img class="d-block w-100" src="/img/logos/sageailabs.png" alt="First slide">
```

File: content/en/_index.html [hunk 2]
```diff
@@ -97,6 +97,54 @@ <h2 class="section-label">GitOps at your fingertips</h2>
...
+          </div>
+          <div class="carousel-item">
+            <img class="d-block w-100" src="http://localhost:1313/img/logos/bimspot.png" alt="Second slide">
```
Comment (dholbach) [inline content/en/_index.html:120]:
```suggestion
            <img class="d-block w-100" src="/img/logos/bimspot.png" alt="Second slide">
```

File: content/en/_index.html [hunk 3]
```diff
@@ -97,6 +97,54 @@ <h2 class="section-label">GitOps at your fingertips</h2>
...
+          </div>
+          <div class="carousel-item">
+            <img class="d-block w-100" src="http://localhost:1313/img/logos/blackmirror-media.png" alt="Third slide">
```
Comment (dholbach) [inline content/en/_index.html:123]:
```suggestion
            <img class="d-block w-100" src="/img/logos/blackmirror-media.png" alt="Third slide">
```

File: content/en/_index.html [hunk 4]
```diff
@@ -97,6 +97,54 @@ <h2 class="section-label">GitOps at your fingertips</h2>
...
+          </div>
+          <div class="carousel-item">
+            <img class="d-block w-100" src="https://fluxcd.io/img/logos/d2iq.png" alt="Third slide">
```
Comment (dholbach) [inline content/en/_index.html:126]:
```suggestion
            <img class="d-block w-100" src="/img/logos/d2iq.png" alt="Third slide">
```

File: content/en/_index.html [hunk 5]
```diff
@@ -97,6 +97,54 @@ <h2 class="section-label">GitOps at your fingertips</h2>
...
+          </div>
+          <div class="carousel-item">
+            <img class="d-block w-100" src="http://localhost:1313/img/logos/adoreme.png" alt="Third slide">
```
Comment (dholbach) [inline content/en/_index.html:129]:
```suggestion
            <img class="d-block w-100" src="/img/logos/adoreme.png" alt="Third slide">
```

**Review by dholbach — CHANGES_REQUESTED** [review]
If you change to relative urls for images they should show up.

**Comment by dholbach** [comment]
Nice, now it's working! Do we want to subtitle the logos with the company name maybe?

**Comment by juozasg** [comment]
Possibly. If the logo already contains the name it could look redundant. I think the source YAML should contain the label (or no label). I will experiment with that and will update the PR.

**Comment by dholbach** [comment]
Looks like it's back to absolute urls for the logos including `localhost:1313`?

**Comment by dholbach** [comment]
One piece of feedback I got was:
> Looks good on my mobile, dominates the screen a bit on a laptop

**Comment by dholbach** [comment]
I'll look into adding a solution for showing company names as caption if necessary.

File: layouts/shortcodes/blocks/carousel.html [hunk 6]
```diff
@@ -11,12 +11,31 @@ <h2 class="text-center section-label">Adopted by</h2>
         <li data-target="#carousel-selected-indicator" data-slide-to="4"></li>
       </ol>
       -->
+      <!-- {{ readFile "/content/en/adopters_carousel_include.html" | safeHTML }} -->
+
       <div class="carousel-inner">
         <a href="/adopters">
-          {{ readFile "/content/en/adopters_carousel_include.html" | safeHTML }}
```
Comment (dholbach) [inline layouts/shortcodes/blocks/carousel.html]: Not sure why this is removed now?

**Comment by dholbach** [comment]
We just got some feedback that maybe a carousel does not show off the breadth of adopters really well - especially if people are on  the landing page for a short while.

I see two options:

- We could try to see if we can get more logos to show in the carousel at the same time?
- We could move to a more static / non-moving kind of layout?

WDYT?

**Comment by dholbach** [comment]
@juozasg In the PR above I changed parts of the script to put the logos in groups of 5 - this doesn't look well yet - maybe you can help?

**Comment by dholbach** [comment]
Another idea: carousel case studies, blogs, endorsements?
- https://www.weave.works/blog/case-study-mettle-leverages-gitops-for-self-service-developer-platform
- https://www.weave.works/blog/case-study-tidepool-streamlines-and-secures-cluster-lifecycle-with-gitops
- https://www.youtube.com/watch?v=59mTx8tBYag
- https://www.youtube.com/watch?v=dEMYVkIU0aU
- https://www.youtube.com/watch?v=ewCnqDgewMg

**Comment by dholbach** [comment]
A new idea I just had:

have our "wall of logos" in the background (opaque), carousel some of the endorsements over them?

**Comment by dholbach** [comment]
Screenshot from 2021-07-14 16-43-08

I worked on something which looks like this right now. The background looks OK-ish. The foreground still needs styling. Happy to get the scripts to generate the right code.

What do people generally feel about this?

**Comment by dholbach** [comment]
I'd cycle e.g. 5 items over the wall of logos, each being left side image, right side some text or quote.

**Comment by scottrigby** [comment]
I personally think that would look great! With a clear link to the adopters page so people can see them all, and see how to add themselves?

**Comment by juozasg** [comment]
Sounds good. I should be able to looks over this tomorrow!

**Comment by dholbach** [comment]
Please take a look at https://github.com/fluxcd/website/pull/413.

**Pull request closed without merging**